src/main/java/uk/gov/hmcts/ccd/domain/types/OrgPolicyCaseAssignedRoleValidator.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.hmcts.ccd.domain.types;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.springframework.beans.factory.annotation.Qualifier;
+import uk.gov.hmcts.ccd.data.caseaccess.CachedCaseRoleRepository;
+import uk.gov.hmcts.ccd.data.caseaccess.CaseRoleRepository;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+@Named
+@Singleton
+public class OrgPolicyCaseAssignedRoleValidator implements FieldIdBasedValidator {
+
+    private final CaseRoleRepository caseRoleRepository;
+
+    @Inject
+    public OrgPolicyCaseAssignedRoleValidator(
+        @Qualifier(CachedCaseRoleRepository.QUALIFIER) final CaseRoleRepository caseRoleRepository
+    ) {
+        this.caseRoleRepository = caseRoleRepository;
+    }
+
+    @Override
+    public String getFieldId() {
+        return PredefinedFieldsIDs.ORG_POLICY_CASE_ASSIGNED_ROLE.getId();
+    }
+
+    @Override
+    public List<ValidationResult> validate(ValidationContext validationContext) {
```

Comment:
Do we need to run this through the `TextValidator` first before the custom validation in this class, otherwise we are losing those basic validations? (i.e. do the same as what `TextCaseReferenceCaseLinkValidator` does)

Reply:
👍

Reply:
well TextCaseReferenceCaseLinkValidator has business requirements that force a relationship with TextValidator. Regular expression, Min and max and others.    OrgPolicyCaseAssignedRoleValidator we only have isEmpty , then the validations are different .  I don't see the benefit of forcing a chain of validations calls and injection only for one validation.